File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Set<Character> set = new HashSet<>();
        int le=0;
        int maxen =0;
        for(int ri =0; ri<s.length(); ri++ ){
            while (set.contains(s.charAt(ri))){
                set.remove(s.charAt(le));
                le++;
            }
            set.add(s.charAt(ri));
            maxen = Math.max(maxen, ri-le+1);
        }
        return maxen;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int left = 0;
        int right = 0;
        int maxlength=0;
        unordered_map<char,int>m;
        int n = s.size();
        for( right = 0; right<n ; right++){
            int ch = s[right];
           m[ch]++;
        
        while(m[ch]>1){
            m[s[left]]--;
            left++;
        }
maxlength = max(maxlength,right-left+1);
    }
    return maxlength;
}
};